DeepSpeech.ipynb

```diff
@@ -1375,26 +1333,37 @@
     "                writer.add_run_metadata(loss_run_metadata, 'loss_epoch%d_batch%d'   % (epoch, batch))\n",
     "            writer.flush()\n",
     "        \n",
-    "        # Print progress message\n",
+    "        # Print WER report\n",
     "        if epoch % display_step == 0:\n",
     "            print \"Epoch:\", '%04d' % (epoch), \"avg_cer=\", \"{:.9f}\".format((total_accuracy / ceil(batches_per_device)))\n",
-    "            _, last_train_wer = calculate_and_print_wer_report( \\\n",
-    "                session, \\\n",
-    "                \"Training\", \\\n",
-    "                tower_decodings, \\\n",
-    "                tower_labels, \\\n",
-    "                tower_total_losses)\n",
+    "            \n",
+    "            # Calculating the training WERs\n",
+    "            rates, train_wer = wers(originals, results)\n",
+    "            \n",
+    "            # Printing the training WER report\n",
+    "            print_wer_report('Training', train_wer, zip(originals, results, rates, losses))\n",
     "            print\n",
     "            \n",
     "        # Validation step\n",
     "        if epoch % validation_step == 0:\n",
-    "            _, last_validation_wer = calculate_and_print_wer_report( \\\n",
-    "                session, \\\n",
-    "                \"Validation\", \\\n",
-    "                validation_tower_decodings, \\\n",
-    "                validation_tower_labels, \\\n",
-    "                validation_tower_total_losses)\n",
-    "            print\n",
+    "            originals = []\n",
```

Comment:
This de-normalization was done on purpose. I already had a merged version and due to all the extra bits of the training run, this was pretty complicated and ugly. However: Merging only validation and test into a separate routine seems to be asymmetric (as in contrast to the in-lined training code). I'll test some other variants for this.